hazelcast/src/main/java/com/hazelcast/sql/SqlQuery.java

```diff
@@ -0,0 +1,289 @@
+
+    /** Default page size. */
+    public static final int DEFAULT_CURSOR_BUFFER_SIZE = 4096;
+
+    private String sql;
+    private List<Object> parameters = new ArrayList<>();
+    private long timeout = DEFAULT_TIMEOUT;
+    private int cursorBufferSize = DEFAULT_CURSOR_BUFFER_SIZE;
+
+    public SqlQuery(@Nonnull String sql) {
+        setSql(sql);
+    }
+
+    /**
+     * Copying constructor.
+     */
+    private SqlQuery(String sql, List<Object> parameters, long timeout, int cursorBufferSize) {
+        this.sql = sql;
+        this.parameters = parameters;
+        this.timeout = timeout;
+        this.cursorBufferSize = cursorBufferSize;
+    }
+
+    /**
+     * Gets the SQL query to be executed.
+     *
+     * @return SQL query
+     */
+    @Nonnull
+    public String getSql() {
+        return sql;
+    }
+
+    /**
+     * Sets the SQL query to be executed.
+     * <p>
+     * The SQL query cannot be null or empty.
+     *
+     * @param sql SQL query
+     * @return this instance for chaining
+     * @throws NullPointerException if passed SQL query is null
+     * @throws IllegalArgumentException if passed SQL query is empty
+     */
+    @Nonnull
+    public SqlQuery setSql(@Nonnull String sql) {
+        Preconditions.checkNotNull(sql, "SQL cannot be null");
+
+        if (sql.length() == 0) {
+            throw new IllegalArgumentException("SQL cannot be empty");
+        }
+
+        this.sql = sql;
+
+        return this;
+    }
+
+    /**
+     * Gets the query parameters.
+     *
+     * @return query parameters
+     */
+    @Nonnull
+    public List<Object> getParameters() {
+        return parameters;
+    }
+
+    /**
+     * Sets the query parameters.
+     * <p>
+     * You may define parameter placeholders in the query with the {@code "?"} character. For every placeholder, a parameter
+     * value must be provided.
+     * <p>
+     * When the method is called, the content of the parameters list is copied. Subsequent changes to the original list don't
+     * change the query parameters.
+     *
+     * @param parameters query parameters
+     * @return this instance for chaining
+     *
+     * @see #addParameter(Object)
+     * @see #clearParameters()
+     */
+    @Nonnull
+    public SqlQuery setParameters(List<Object> parameters) {
+        if (parameters == null || parameters.isEmpty()) {
+            this.parameters = new ArrayList<>();
+        } else {
+            this.parameters = new ArrayList<>(parameters);
+        }
+
+        return this;
+    }
+
+    /**
+     * Adds a single parameter to the end of the parameters list.
+     *
+     * @param parameter parameter
+     * @return this instance for chaining
+     *
+     * @see #setParameters(List)
+     * @see #clearParameters()
+     */
+    @Nonnull
+    public SqlQuery addParameter(Object parameter) {
+        parameters.add(parameter);
+
+        return this;
+    }
+
+    /**
+     * Clear query parameters.
```

Comment:
Minor: "Clears query parameters" to be consistent with the rest of the JavaDoc

Reply:
Fixed.